src/core/common_conditions.rs

```diff
@@ -0,0 +1,76 @@
+use bevy::prelude::*;
+
+use crate::{client::replicon_client::RepliconClient, server::replicon_server::RepliconServer};
+
+/// Returns `true` if the server is running.
+pub fn server_running(server: Option<Res<RepliconServer>>) -> bool {
+    server.filter(|server| server.is_running()).is_some()
+}
+
+/// Returns `true` if there is no client or if the existing client is disconnected.
+///
+/// Can be used for systems that run both on the server and in singleplayer mode.
+pub fn has_authority(client: Option<Res<RepliconClient>>) -> bool {
+    let Some(client) = client else {
+        return true;
+    };
+    client.is_disconnected()
+}
+
+/// Returns `true` when the client is connecting.
+pub fn client_connecting(client: Option<Res<RepliconClient>>) -> bool {
+    client.filter(|client| client.is_connecting()).is_some()
+}
+
+/// Returns `true` when the client is connected.
+pub fn client_connected(client: Option<Res<RepliconClient>>) -> bool {
+    client.filter(|client| client.is_connected()).is_some()
+}
+
+/// Returns `true` if the server stopped on this tick.
```

Comment:
@UkoeHB maybe clarity that it's not exactly on this tick, but more like since the condition run? Because if you run this condition again on the same "stopped" tick, it will return `false`.

Reply:
Ah, it's Bevy's tick. I assumed that we meant replicon tick as in other places.